src/main/java/apim/restful/importexport/utils/APIImportUtil.java

```diff
@@ -180,100 +180,91 @@ public static String extractArchive(File sourceFile, String destination) throws
+     * @throws APIImportException     if there is an error in importing an API
      */
     public static void importAPI(String pathToArchive, String currentUser, boolean isDefaultProviderAllowed)
-            throws APIManagementException, APIImportException {
+            throws APIImportException {
 
         API importedApi;
 
+        // If the original provider is preserved,
         if (isDefaultProviderAllowed) {
-            importedApi = preserveOriginalProvider(pathToArchive);
+
+            FileInputStream inputStream = null;
+            BufferedReader bufferedReader = null;
+
+            try {
+                inputStream = new FileInputStream(pathToArchive + APIImportExportConstants.JSON_FILE_LOCATION);
+                bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+                importedApi = new Gson().fromJson(bufferedReader, API.class);
+            } catch (FileNotFoundException e) {
+                log.error("Error in locating api.json file. ", e);
+                throw new APIImportException("Error in locating api.json file. " + e.getMessage());
+            } finally {
+                IOUtils.closeQuietly(inputStream);
+                IOUtils.closeQuietly(bufferedReader);
+            }
         } else {
-            importedApi = setApiProviderToCurrentUser(pathToArchive + APIImportExportConstants.JSON_FILE_LOCATION,
-                    currentUser);
+
+            String pathToJSONFile = pathToArchive + APIImportExportConstants.JSON_FILE_LOCATION;
+
+            try {
+                String jsonContent = FileUtils.readFileToString(new File(pathToJSONFile));
+                JsonElement configElement = new JsonParser().parse(jsonContent);
+                JsonObject configObject = configElement.getAsJsonObject();
+
+                //locate the "providerName" within the "id" and set it as the current user
+                JsonObject apiId = configObject.getAsJsonObject(APIImportExportConstants.ID_ELEMENT);
+                apiId.addProperty(APIImportExportConstants.PROVIDER_ELEMENT, APIUtil.replaceEmailDomain(currentUser));
+                importedApi = new Gson().fromJson(configElement, API.class);
+
+            } catch (IOException e) {
+                log.error("Error in setting API provider to logged in user. ", e);
+                throw new APIImportException("Error in setting API provider to logged in user. " + e.getMessage());
+            }
         }
 
-        Set<Tier> allowedTiers = provider.getTiers();
-        boolean isAllTiersAvailable = allowedTiers.containsAll(importedApi.getAvailableTiers());
+        Set<Tier> allowedTiers;
+        Set<Tier> unsupportedTiersList;
 
-        if (!isAllTiersAvailable) {
+        try{
+            allowedTiers = provider.getTiers();
+        } catch (APIManagementException e) {
+            log.error("Error in retrieving tiers of the provider. ", e);
```

Comment:
Has been fixed in commit : https://github.com/wso2/product-apim/commit/995f737a32616db5a1ef8950a5d4618f060b530f